source/renderer/zettlr-editor.js

```diff
@@ -222,6 +222,39 @@ class ZettlrEditor {
 
     this._searcher.setInstance(this._cm)
 
+    this._cm.on('keyup', (cm, event) => {
```

Comment:
This whole logic can be done much easier with a CodeMirror shortcut. Have a look at the CodeMirror manual and at the file `generate-keymap.js` (resides in `source/renderer/assets/codemirror`) to get an idea of where to put this logic!